### PR TITLE
Use major version of GitHub Actions

### PR DESCRIPTION
**Pull request by basil**
See https://github.com/jenkinsci/file-parameters-plugin/pull/90#discussion_r835918340 - even though this PR subjects us to https://github.com/dependabot/dependabot-core/issues/4834 I think it is still preferable to the status quo, wherein we get lots of noise like #6426.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]
+1 for less noise.

File: .github/workflows/publish-release-artifact.yml
Comment (NotMyFault) [inline .github/workflows/publish-release-artifact.yml:61]: Is there are particular reason why core uses such an ancient version? The repository providing it is EOL since early 2021, therefore pinning this specific version has bar no effect on it.

**Review by jglick — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by MarkEWaite — APPROVED** [review]
Great for me.  Hopefully the dependabot issue will be fixed so that major version upgrades will be proposed when they happen.  Until the bug is fixed, we can rely on a few of the plugin repositories that will likely continue using the fully specified version number.  They can be the "canary" to alert a few people when a new major version is released and dependabot doesn't notify us.

**Review by timja — APPROVED** [review]

**Pull request merged**
